### ISSUE TITLE
how to set lambda properly?

### ISSUE THREAD

**Issue by DISAPPEARED13** [junyanz/pytorch-CycleGAN-and-pix2pix#1325]
I see the paper set identity loss as 0.5lambda and cycle-consistent loss as lambda to control the relative importance. What if GAN loss just about 0.5-1.0 around but idt loss and cycle loss up to at least 10 at the first. Will it be ignoring the adversarial loss while training?
thanks alot. : )
by the way... when i change the framework to 3d, my gpu memory ran out and I exchange to torch.cuda.amp mode, with gradscaler(), will it make the generator performance worse?
I use it as medical image generating and my MAE just around 90... BAD performance... : (

**Comment by DISAPPEARED13** [comment]
@junyanz plz, thanks alot. this disturbs me for such a long time..!

**Comment by junyanz** [comment]
You might want to change lambda if you use a different GAN loss as each loss has a different range.

**Comment by DISAPPEARED13** [comment]
> You might want to change lambda if you use a different GAN loss as each loss has a different range.

Thanks alot!

**Comment by youngprogrammerBee** [comment]
> I see the paper set identity loss as 0.5lambda and cycle-consistent loss as lambda to control the relative importance. What if GAN loss just about 0.5-1.0 around but idt loss and cycle loss up to at least 10 at the first. Will it be ignoring the adversarial loss while training? thanks alot. : ) by the way... when i change the framework to 3d, my gpu memory ran out and I exchange to torch.cuda.amp mode, with gradscaler(), will it make the generator performance worse? I use it as medical image generating and my MAE just around 90... BAD performance... : (

Hi and sorry to interrupt, I am also working on generating 3d medical image, but I found some difficulties in adjusting the net framework. Did you also generated 3d results that represented by lots of images ?

**Comment by DISAPPEARED13** [comment]
> > I see the paper set identity loss as 0.5lambda and cycle-consistent loss as lambda to control the relative importance. What if GAN loss just about 0.5-1.0 around but idt loss and cycle loss up to at least 10 at the first. Will it be ignoring the adversarial loss while training? thanks alot. : ) by the way... when i change the framework to 3d, my gpu memory ran out and I exchange to torch.cuda.amp mode, with gradscaler(), will it make the generator performance worse? I use it as medical image generating and my MAE just around 90... BAD performance... : (
> 
> Hi and sorry to interrupt, I am also working on generating 3d medical image, but I found some difficulties in adjusting the net framework. Did you also generated 3d results that represented by lots of images ?

Sure. What's your problem? I used to stuck in here for much long time.

**Comment by youngprogrammerBee** [comment]
> 

Thanks for your reply, I wonder if you could please show your changed cycle-gan code generating 3d images on your github? I found it difficult to change the code to generate 3d images for a deep learning beginner like me, but i really want to watch the results on cycle-gan generating 3d images.

**Comment by youngprogrammerBee** [comment]
> > > I see the paper set identity loss as 0.5lambda and cycle-consistent loss as lambda to control the relative importance. What if GAN loss just about 0.5-1.0 around but idt loss and cycle loss up to at least 10 at the first. Will it be ignoring the adversarial loss while training? thanks alot. : ) by the way... when i change the framework to 3d, my gpu memory ran out and I exchange to torch.cuda.amp mode, with gradscaler(), will it make the generator performance worse? I use it as medical image generating and my MAE just around 90... BAD performance... : (
> > 
> > 
> > Hi and sorry to interrupt, I am also working on generating 3d medical image, but I found some difficulties in adjusting the net framework. Did you also generated 3d results that represented by lots of images ?
> 
> Sure. What's your problem? I used to stuck in here for much long time.

Thanks for your reply, I wonder if you could please show your changed cycle-gan code generating 3d images on your github? I found it difficult to change the code to generate 3d images for a deep learning beginner like me, but i really want to watch the results on cycle-gan generating 3d images.
